cli/pcluster/config/pcluster_config.py

```diff
@@ -296,6 +321,46 @@ def __init_section_from_file(self, section_definition, config_parser, section_la
         except configparser.NoSectionError as e:
             self.error("Section '[{0}]' not found in the config file.".format(e.section))
 
+    def __refresh_parameters(self):
+        """
+        Refresh all parameters.
+
+        Invokes the refresh() method of all parameters, giving them the opportunity for updating their value based on
+        the PclusterConfig status.
+        """
+        for _, sections in self.__sections.items():
+            for _, section in sections.items():
+                for _, param in section.params.items():
+                    param.refresh()
+
+    def __refresh(self):
+        """
+        Reload the sections structure.
+
+        Reloading is performed only if the configuration has been marked as stale (in general by invoking the
+        public refresh() method) and only if a refresh operation is not already in progress.
+        """
+        if self.__stale and not self.__refreshing:
+            self.__refreshing = True
+            new_sections = OrderedDict({})
+            for key, sections in self.__sections.items():
+                new_sections_map = {}
+                for _, section in sections.items():
+                    new_sections_map[section.label] = section
+                new_sections[key] = new_sections_map
+            self.__sections = new_sections
+            self.__refresh_parameters()
+            self.__stale = False
+            self.__refreshing = False
+
+    def refresh(self):
+        """
+        Mark the object as stale.
+
+        When the object is marked as stale, the next call to one of its public methods will trigger a refresh operation.
+        """
+        self.__stale = True
```

Comment:
see my previous comment about internal state.
this method could internally call a method `_change_state(event.REFRESH_REQUIRED)` to change the internal state to `` STALE`, according to the prvious state and the event

Reply:
Why calling this additional method and not just setting the status to STALE?